docs/user-documentation.rst

```diff
@@ -94,103 +94,57 @@ And one that does not need ``nested=True``::
-+---------------------------------------+--------+-------------------------------------------+
-| ``SWITCHBOARD_ROOT/status``           | JSON   | NA                                        |
-+---------------------------------------+--------+-------------------------------------------+
-| ``SWITCHBOARD_ROOT/delete``           | JSON   | NA                                        |
-+---------------------------------------+--------+-------------------------------------------+
-| ``SWITCHBOARD_ROOT/add_condition``    | JSON   | NA                                        |
-+---------------------------------------+--------+-------------------------------------------+
-| ``SWITCHBOARD_ROOT/remove_condition`` | JSON   | NA                                        |
-+---------------------------------------+--------+-------------------------------------------+
-| ``SWITCHBOARD_ROOT/history``          | JSON   | NA                                        |
-+---------------------------------------+--------+-------------------------------------------+
-
-For more details, please look through the example_ code. Once the views are
-defined switches may be used in the code.
-
-Post-Request Cleanup
-^^^^^^^^^^^^^^^^^^^^
-
-The last thing to setup is to trigger an event when the request is finished.
-Switchboard needs to cleanup some caching data. If this event is not triggered
-changes to the switches will not propogate out without server restarts.
-Depending on the framework's architecture invoking something at the end of a
-request may mean creating some sort of WSGI middleware or implementing an
-event handler. For example, as WSGI middleware::
-
-    from webob import Request
-    from switchboard.signals import request_finished
-
-    class SwitchboardMiddleware(object):
-
-        def __init__(self, app):
-            self.app = app
-
-        def __call__(self, environ, start_response):
-            req = resp = None
-            try:
-                req = Request(environ)
-                resp = req.get_response(self.app)
-                return resp(environ, start_response)
-            finally:
-                self._end_request(req)
-
-        def _end_request(self, req):
-            if req:
-                # Notify Switchboard that the request is finished
-                request_finished.send(req)
+Middleware
+^^^^^^^^^^
 
+The last thing to setup is to handle pre- and post-request tasks. Pre-request
+tasks can include adding objects to the context (eliminating the need to add
+them explicitly when querying ``is_active``). Post-request tasks include
+cleaning up caching data once a request is finished. Switchboard includes
+middleware to handle these tasks. Using it out of the box::
+
+    from switchboard.middlware import SwitchboardMiddlware
+    app = SwitchboardMiddleware(app)
+
+It can also be extended for further customization, specifically by implementing
+the ``pre_request`` method. For example, to add a user object to the context::
```

Comment:
I do like the idea of it though, so that `user` and `request` can be inserted more cleanly into the context, compared to before.